language/stdlib/modules/libra_system.move

```diff
@@ -186,69 +196,186 @@ module LibraSystem {
         abort 99
     }
 
+    // Get the index of the discovery info with address `addr` in `discovery_set`.
+    // Aborts if `addr` is not in discovery set.
+    public fun get_discovery_index(discovery_set: &vector<DiscoveryInfo>, addr: address): u64 {
+        let len = Vector::length(discovery_set);
+        let i = 0;
+        loop {
+            if (get_discovery_address(Vector::borrow(discovery_set, i)) == &addr) {
+                return i
+            };
+
+            i = i + 1;
+            if (i >= len) break;
+        };
+
+        abort 99
+    }
+
     // Adds a validator to the addition buffer, which will cause it to be added to the validator
     // set in the next epoch.
     // Fails if `account_address` is already a validator or has already been added to the addition
     // buffer.
-    // Only callable by the Association address
-    public fun add_validator(account_address: address) acquires ValidatorSet {
+    // Only callabcle by the Association address
```

Comment:
```suggestion
    // Only callable by the Association address
```

Reply:
What enforces this?

Reply:
The implementation `add_validator_` will have an assertion. Not sure if we need to duplicate the assertion there or we may specify it via move prover in the future.

Reply:
Ideally, this would take a `&AssociationCapability` and then pass it to `add_validator_` (@tzakian may know more about assoc address -> assoc capability migration?)